src/generate.js

```diff
@@ -246,44 +244,43 @@ const runStringHandlers = (
  */
 export const generateCSSRuleset = (
     selector /* : string */,
-    declarations /* : SheetDefinition */,
+    declarations /* : OrderedElements */,
     stringHandlers /* : StringHandlers */,
     useImportant /* : boolean */,
     selectorHandlers /* : SelectorHandler[] */
 ) /* : string */ => {
-    const handledDeclarations = runStringHandlers(
+    const handledDeclarations /* : OrderedElements */ = runStringHandlers(
         declarations, stringHandlers, selectorHandlers);
 
-    const prefixedDeclarations = prefixAll(handledDeclarations);
+    const prefixedDeclarations = prefixAll(handledDeclarations.elements);
 
     const prefixedRules = flatten(
         objectToPairs(prefixedDeclarations).map(([key, value]) => {
             if (Array.isArray(value)) {
                 // inline-style-prefix-all returns an array when there should be
                 // multiple rules, we will flatten to single rules
-
-                const prefixedValues = [];
-                const unprefixedValues = [];
-
-                value.forEach(v => {
-                  if (v[0] === '-') {
-                    prefixedValues.push(v);
-                  } else {
-                    unprefixedValues.push(v);
-                  }
-                });
-
-                prefixedValues.sort();
-                unprefixedValues.sort();
-
-                return prefixedValues
-                  .concat(unprefixedValues)
-                  .map(v => [key, v]);
+                return value.map(v => [key, v]);
             }
             return [[key, value]];
         })
     );
 
+    // Calculate the order that we want to each element in `prefixedRules` to
+    // be in, based on its index in the original key ordering.
+    const sortOrder = {};
+    for (let i = 0; i < prefixedRules.length; i++) {
+        const key = prefixedRules[i][0];
+        sortOrder[key] = handledDeclarations.keyOrder.indexOf(key);
+    }
+
+    // Sort the prefixed rules according to the order that the keys were
+    // in originally before we prefixed them. Since the new, prefixed versions
+    // of the rules aren't in the original list, their sort order will be -1
+    // and we sort them to the top.
+    prefixedRules.sort((a, b) => {
+        return sortOrder[a[0]] - sortOrder[b[0]];
+    });
```

Comment:
I think that might be okay as long as we only do this for prefixed styles that do not have a sortOrder. In other words, if I explicitly pass in a prefixed style after its unprefixed style, that order should be maintained.